### PR TITLE
Add PSU state sensor support for ArubaOS-CX & add serial and firmware

### PR DESCRIPTION
**Pull request by rudybroersma**
This is a pull request to add:
1. PSU (Power Supply) state sensor support
2. Add additional OIDs for serial number and software/firmware version

Addition 1:
Currently, LibreNMS does not have a state sensor to monitor the power supply status (ok/fail).
When I pull the power cord on one of the PSUs the operStatus changes to nonoperational. 

Addition 2:

The current discovery yaml expects the serial number to be at:
ENTITY-MIB::entPhysicalSerialNum.1

All Aruba's I have (6200s and 6300s) have the serial number at
ENTITY-MIB::entPhysicalSoftwareRev.101001

I believe the difference has to do with the fact that atleast the 6000-series seem to believe the device itself is just a container and the switch is a virtual (?) blade, identified as 101001. In reality though it's just a simple 1U fixed config device.

snmpsim test data added. ./lnms dev:check succeeds. Added a little screenshot of the new state sensor as an example:

aruba-state-sensor

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [X] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [X] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by rudybroersma** [comment]
Currently I have this OID defined statically:

$arubaPowerSensorOID = "enterprises.47196.4.1.1.2.1.3"; // ARUBAWIRED-NETWORKING-OID::arubaWiredPowerSensor

Would it be better to retrieve this value from the MIB using snmp_get?
I chose not to do that, as it would imply I also need to add the MIB files into LibreNMS, and I wasn't sure about copyrights

**Comment by mpikzink** [comment]
Nice, I just want to suggest naming the sensor values consistently. Once you wrote it in small letters and once in capital letters.

I would even suggest using just Power Supply x/x as the name both times. The type makes it clear enough, it would be cleaner.

**Comment by rudybroersma** [comment]
> Nice, I just want to suggest naming the sensor values consistently. Once you wrote it in small letters and once in capital letters.
> 
> I would even suggest using just Power Supply x/x as the name both times. The type makes it clear enough, it would be cleaner.

I didn't name them. The description is coming from the device itself. 
It gets it in this line:

$descr = snmp_get($device, 'ENTITY-MIB::entPhysicalName.'.$index, '-Ovqe', 'ENTITY-MIB');

The two Power sensors are not mine. They are detected in the physical-entity code. They haven't changed by me. I just added the two state sensors. Sorry if that was unclear.

**Comment by PipoCanaja** [comment]
Hi @rudybroersma

Thanx for this PR which is very complete. 

We usually prefer the YAML form for discovery instead of PHP (more compact, easier to maintain) and you can of course add the MIB file. They don't contain any data, only a translation from numbers to name, so there is not much to protect here. And you add only the necessary MIB files, not all of them. 

If you prefer to stay like this, let us know.

**Comment by rudybroersma** [comment]
Hi @PipoCanaja 

I looked at that, but I don't think it's possible to do this in YAML? If you feel I'm missing something please let me know.

The reason I think it's not possible to do with this 'just'  YAML is because these sensors are in the list of physical entity sensors. I need to iterate through the list of entity sensors, check if the sensor is of the proper type (line 29 in [arubaos-cx.inc.php](https://github.com/librenms/librenms/pull/15687/commits/fa8e1af5f503e2f3b49f4a2c79766805db0ab612#diff-1db85e8c3f4dcbcf381c0483f07014bb08ebeebf95e2c1e96aa6f32de498defb) (if sensor arubaWIredPowerSensor?) and then add the sensor to LibreNMS. 

In other words, these state sensors do not have their own table

I don't think it's possible to do this 'if then'  clause using YAML only? If you feel it is possible, could you point me to an example? I'll gladly rewrite the patch.

Additionally, if it's okay to add the MIBs I'll also update the PR to include the MIBs and change line 16 of arubaos-cx.inc.php (or YAML)  to dynamically look up the OID for arubaWiredPowerSensor sensor type.

**Comment by murrant** [comment]
@rudybroersma have you tried enabling the entity-physical module for this device?

**Comment by rudybroersma** [comment]
> @rudybroersma have you tried enabling the entity-physical module for this device?

That module is on by default for this device (at least on my installation)

**Comment by rudybroersma** [comment]
Managed to fix this using YAML. Closing this one.
